### PR TITLE
Defcon release

### PR DESCRIPTION
**Pull request by psc4re**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,36 +24,80 @@ Latest version of Subfinder installed, for instructions see https://github.com/p
...
-Graylog2 requires a few steps to start receiving logs from ASF: 
+`Note that if social login is enabled and not configured in admin panel ASF will not display the login page - to revert simply disable the Social Login in .env.prod`
 
-Once logged in, go to System/"Content Packs" and import the Content Pack located at /opt/asf/tools/graylog/content_pack_ASF.json, click on the "Upload" button and you should see "Basic" reflected in the "Select Content Packs" section, click on "Basic", make sure the "ASF" radio button is selected and hit the "Apply content" button, this will create the Global input to parse JSON logs and related extractors. 
 
-![Graylog2 Inputs Example](images/Graylog_content_pack.jpg)
+###### Jira Integration
 
-Now you are ready to receive logs from ASF and setup your streams / alerts / dasboards ! 
+ASF support Jira integration and ticketing to enable Jira set the following values in .env.prod:
```
Comment (pavsorab) [inline README.md:86]: My suggestion "To enable Jira integration and ticketing, ASF supports setting the following values in the .env.prod file"

File: frontend/asfui/app/management/commands/Jira_Conf.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    username = urllib.parse.quote_plus(settings.MONGO_USER)
+    password = urllib.parse.quote_plus(settings.MONGO_PASSWORD)
+    url = settings.MONGO_URL
+    port = settings.MONGO_PORT
+    client = pymongo.MongoClient(f"mongodb://{username}:{password}@{url}:{port}")
+    myDB = client["Nuclei"]
+    mycol = myDB["report"]
+    mycol.update_many(query, {"$set":{"verified":True, "ignored":True}})
+    mycol.count_documents(query)
+    print("ignored issue", query, mycol.count_documents(query))
```
Comment (pavsorab) [inline frontend/asfui/app/management/commands/Jira_Conf.py:71]: Please consider removing `print` statements recommend using debug logger.

File: frontend/asfui/app/management/commands/Jira_Conf.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+def main():
+    create_issue()
+    print("issue created")
```
Comment (pavsorab) [inline frontend/asfui/app/management/commands/Jira_Conf.py:75]: Please consider removing `print` statements recommend using debug logger.

File: frontend/asfui/app/management/commands/Jira_Conf.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+class Command(BaseCommand):
+    def add_arguments(self, parser):
+        parser.add_argument("--ticket", type=str, required=True)
+
+    def handle(self, *args, **kwargs):
+        severity = kwargs['ticket']
+        try:
+            query = {"info.severity":severity}
+            create_issue(query)
+            print("insert complete")
```
Comment (pavsorab) [inline frontend/asfui/app/management/commands/Jira_Conf.py:89]: Please consider removing `print` statements recommend using debug logger.

File: frontend/asfui/app/management/commands/Jira_Conf.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        except Exception as e:
+            print("insert failed"+ str(e))
```
Comment (pavsorab) [inline frontend/asfui/app/management/commands/Jira_Conf.py:91]: Please consider removing `print` statements recommend using error logger.

File: frontend/asfui/app/management/commands/remaster_output.py [hunk 6]
```diff
@@ -193,6 +196,80 @@ def parser_nuclei_http(PARSER_INPUT, PARSER_OUTPUT):
             debug("Found nothing:"+line)
     return
 
+
+def parser_wpscan_json(PARSER_INPUT, PARSER_OUTPUT):
+    print("Sending to parser for WPScan")
```
Comment (pavsorab) [inline frontend/asfui/app/management/commands/remaster_output.py:201]: Please consider removing `print` statements recommend using debug logger.

File: frontend/asfui/app/management/commands/remaster_output.py [hunk 7]
```diff
@@ -193,6 +196,80 @@ def parser_nuclei_http(PARSER_INPUT, PARSER_OUTPUT):
...
+                row = {'host': host, 'jira_created': False, 'verified':False, "ignored":False, 'info':{'name':i['title'], 'description': i['title'], 'severity': severity}, 'matched-at':file_data['main_theme']['location'], 'template-id': i['title'], 'timestamp':timestamp }
+                result_list.append(row) 
+        for i in file_data["plugins"]:
+            if "vulnerabilities" in file_data["plugins"][i]:
+                    for j in file_data["plugins"][i]["vulnerabilities"]:
+                        if "title" in j:
+                            row = {'host': host, 'jira_created': False, 'verified':False, "ignored":False, 'info':{'name':j['title'], 'description': j['title'], 'severity': severity}, 'matched-at':file_data["plugins"][i]['location'], 'template-id': j['title'], 'timestamp':timestamp }
+                            result_list.append(row) 
+    except Exception as e:
+        print("error in parsing WPscan report:", str(e))
```
Comment (pavsorab) [inline frontend/asfui/app/management/commands/remaster_output.py:234]: Please consider removing `print` statements recommend using error logger.

File: frontend/asfui/app/management/commands/remaster_output.py [hunk 8]
```diff
@@ -193,6 +196,80 @@ def parser_nuclei_http(PARSER_INPUT, PARSER_OUTPUT):
...
+        pass
+        
+    for x in result_list:
+        primaryBulkArr.append(pymongo.UpdateOne({"template-id": x['template-id'], 'host':x['host'], 'matched-at':x['matched-at']}, {'$set':x, '$inc':{'counter':1}}, upsert=True))
+    if len(primaryBulkArr)> 0:
+        Collection.bulk_write(primaryBulkArr)
+    else:
+        print("nothing to insert")
```
Comment (pavsorab) [inline frontend/asfui/app/management/commands/remaster_output.py:242]: Please consider removing `print` statements recommend using debug/error logger.

File: frontend/asfui/app/management/commands/remaster_output.py [hunk 9]
```diff
@@ -193,6 +196,80 @@ def parser_nuclei_http(PARSER_INPUT, PARSER_OUTPUT):
...
+    for index,x in enumerate(file_data):
+        if Collection.count_documents({"template-id": x['template-id'], 'host': x['host'], 'matched-at':x['matched-at']}) < 1:
+            x.update({'jira_created':False, 'verified':False, 'ignored':False})
+            primaryBulkArr.append(pymongo.UpdateOne({"template-id": x['template-id'], 'host':x['host'], 'matched-at':x['matched-at']}, {'$set':x, '$inc':{'counter':1}}, upsert=True))
+        else:
+            primaryBulkArr.append(pymongo.UpdateOne({"template-id": x['template-id'], 'host':x['host'], 'matched-at':x['matched-at']}, {'$set':x, '$inc':{'counter':1}}, upsert=True))
+    if len(primaryBulkArr)> 0:
+        Collection.bulk_write(primaryBulkArr)
+    else:
+        print("nothing to insert")
```
Comment (pavsorab) [inline frontend/asfui/app/management/commands/remaster_output.py:268]: Please consider removing `print` statements recommend using debug/error logger.

File: frontend/asfui/app/management/commands/remaster_output.py [hunk 10]
```diff
@@ -193,6 +196,80 @@ def parser_nuclei_http(PARSER_INPUT, PARSER_OUTPUT):
...
+    
+    return 
+    
+
+
+def parser_nuclei_json(PARSER_INPUT, PARSER_OUTPUT):
+    username = urllib.parse.quote_plus(settings.MONGO_USER)
+    password = urllib.parse.quote_plus(settings.MONGO_PASSWORD)
+    url = settings.MONGO_URL
```
Comment (pavsorab) [inline frontend/asfui/app/management/commands/remaster_output.py:251]: Same comment as suggested in https://github.com/vmware-labs/attack-surface-framework/pull/18/files#diff-ccb2b112e818f612867f84573df1c02da2d80a2832dcda026b492da103e85078R45

File: frontend/asfui/app/views.py [hunk 11]
```diff
@@ -174,6 +180,91 @@ def target_delete():
...
+def alertspage(request):
+    context = {}
+    context['segment'] = 'alerts'
+    global PARSER_DEBUG
+    PARSER_DEBUG=True
+    debug(request.POST)
+    m_query = {"ignored":False}
+    mongo_username = urllib.parse.quote_plus(settings.MONGO_USER)
+    mongo_password = urllib.parse.quote_plus(settings.MONGO_PASSWORD)
+    mongo_url = settings.MONGO_URL
```
Comment (pavsorab) [inline frontend/asfui/app/views.py:197]: Same comment as suggested in https://github.com/vmware-labs/attack-surface-framework/pull/18/files#diff-ccb2b112e818f612867f84573df1c02da2d80a2832dcda026b492da103e85078R45

File: setup.sh [hunk 12]
```diff
@@ -50,6 +50,45 @@ for file in *
...
+
+if [ $localdb == "y" ]
+then 
+	echo "Username for DB" 
+	read mdb_user
+	echo "MONGO_USER=$mdb_user" | tee -a .env.prod
+	echo "Password:" 
+	read -s mdb_pass
+	echo "MONGO_USER=$mdb_pass" | tee -a .env.prod
+	echo "MONGO_URL=10.188.52.96\nMONGO_PORT=27017" | tee -a .env.prod
```
Comment (pavsorab) [inline setup.sh:65]: Please refrain from using private IP addresses.

File: frontend/asfui/authentication/views.py [hunk 13]
```diff
@@ -30,9 +30,8 @@ def login_view(request):
             else:    
                 msg = 'Invalid credentials'
```
Comment (pavsorab) [inline frontend/asfui/authentication/views.py:31]: Please consider adding some error loggers here.

File: frontend/asfui/authentication/views.py [hunk 14]
```diff
@@ -30,9 +30,8 @@ def login_view(request):
             else:    
                 msg = 'Invalid credentials'    
         else:
-            msg = 'Error validating the form'    
-
-    return render(request, "accounts/login.html", {"form": form, "msg" : msg})
+            msg = 'Error validating the form'
```
Comment (pavsorab) [inline frontend/asfui/authentication/views.py:33]: Please consider adding some error loggers here.

File: frontend/asfui/core/settings.py [hunk 15]
```diff
@@ -101,6 +173,200 @@
...
+    SAML_ATTRIBUTE_MAPPING = {
+	    'NameID': ('username',),
+        'email': ('emailAddress', ),
+        'Firstname': ('Firstname', ),
+        'Lastname': ('Lastname', ),
+        }
+    
+    from os import path
+    ASF_URL = "http://127.0.0.1:8000"
+    SSO_URL = 'https://myvmware.workspaceair.com'
```
Comment (pavsorab) [inline frontend/asfui/core/settings.py:251]: Please avoid using any organization-specific internal domains. Please make them configurable for ASF users.

File: frontend/asfui/core/static/assets/css/material-dashboard.css [hunk 16]
```diff
@@ -18823,3 +18836,323 @@ Copyright (c) 2015 Daniel Eden
...
+  -webkit-transition: all .3s ease;
+  -moz-transition: all .3s ease;
+  -ms-transition: all .3s ease;
+  -o-transition: all .3s ease;
+  transition: all .3s ease;
+}
+
+
+#facebook-connect {
+  background: rgb(255, 255, 255) url('https://raw.githubusercontent.com/eswarasai/social-login/master/img/facebook.svg?sanitize=true') no-repeat scroll 5px 0px / 30px 50px padding-box border-box;
```
Comment (pavsorab) [inline frontend/asfui/core/static/assets/css/material-dashboard.css:18892]: It is not recommended to use images from github accounts which are not managed by us. The possible issues could be :

- License issues
- Any maintainers can upload malicious code to images
- maintenance issue (What if user deletes those images?)

Please add required images to our ASF repo.

File: frontend/asfui/core/static/assets/css/material-dashboard.css [hunk 17]
```diff
@@ -18823,3 +18836,323 @@ Copyright (c) 2015 Daniel Eden
...
+  border: 1px solid rgb(60, 90, 154);
+}
+
+#facebook-connect:hover {
+  border-color: rgb(60, 90, 154);
+  background: rgb(60, 90, 154) url('https://raw.githubusercontent.com/eswarasai/social-login/master/img/facebook-white.svg?sanitize=true') no-repeat scroll 5px 0px / 30px 50px padding-box border-box;
```
Comment (pavsorab) [inline frontend/asfui/core/static/assets/css/material-dashboard.css:18898]: It is not recommended to use images from github accounts which are not managed by us. The possible issues could be :

- License issues
- Any maintainers can upload malicious code to images
- maintenance issue (What if user deletes those images?)

Please add required images to our ASF repo.

File: frontend/asfui/core/static/assets/css/material-dashboard.css [hunk 18]
```diff
@@ -18823,3 +18836,323 @@ Copyright (c) 2015 Daniel Eden
...
+  color: #FFF;
+  -webkit-transition: all .3s ease;
+  -moz-transition: all .3s ease;
+  -ms-transition: all .3s ease;
+  -o-transition: all .3s ease;
+  transition: all .3s ease;
+}
+
+#google-connect {
+  background: rgb(255, 255, 255) url('https://raw.githubusercontent.com/eswarasai/social-login/master/img/google-plus.png') no-repeat scroll 5px 0px / 50px 50px padding-box border-box;
```
Comment (pavsorab) [inline frontend/asfui/core/static/assets/css/material-dashboard.css:18931]: It is not recommended to use images from github accounts which are not managed by us. The possible issues could be :

- License issues
- Any maintainers can upload malicious code to images
- maintenance issue (What if user deletes those images?)

Please add required images to our ASF repo.

File: frontend/asfui/core/static/assets/css/material-dashboard.css [hunk 19]
```diff
@@ -18823,3 +18836,323 @@ Copyright (c) 2015 Daniel Eden
...
+  border: 1px solid rgb(220, 74, 61);
+}
+
+#google-connect:hover {
+  border-color: rgb(220, 74, 61);
+  background: rgb(220, 74, 61) url('https://raw.githubusercontent.com/eswarasai/social-login/master/img/google-plus-white.png') no-repeat scroll 5px 0px / 50px 50px padding-box border-box;
```
Comment (pavsorab) [inline frontend/asfui/core/static/assets/css/material-dashboard.css:18937]: It is not recommended to use images from github accounts which are not managed by us. The possible issues could be :

- License issues
- Any maintainers can upload malicious code to images
- maintenance issue (What if user deletes those images?)

Please add required images to our ASF repo.

File: frontend/asfui/core/static/assets/css/material-dashboard.css [hunk 20]
```diff
@@ -18823,3 +18836,323 @@ Copyright (c) 2015 Daniel Eden
...
+  color: #FFF;
+  -webkit-transition: all .3s ease;
+  -moz-transition: all .3s ease;
+  -ms-transition: all .3s ease;
+  -o-transition: all .3s ease;
+  transition: all .3s ease;
+}
+
+#twitter-connect {
+  background: rgb(255, 255, 255) url('https://raw.githubusercontent.com/eswarasai/social-login/master/img/twitter.png') no-repeat scroll 5px 1px / 45px 45px padding-box border-box;
```
Comment (pavsorab) [inline frontend/asfui/core/static/assets/css/material-dashboard.css:18970]: It is not recommended to use images from github accounts which are not managed by us. The possible issues could be :

- License issues
- Any maintainers can upload malicious code to images
- maintenance issue (What if user deletes those images?)

Please add required images to our ASF repo.

File: frontend/asfui/core/static/assets/css/material-dashboard.css [hunk 21]
```diff
@@ -18823,3 +18836,323 @@ Copyright (c) 2015 Daniel Eden
...
+  border: 1px solid rgb(85, 172, 238);
+}
+
+#twitter-connect:hover {
+  border-color: rgb(85, 172, 238);
+  background: rgb(85, 172, 238) url('https://raw.githubusercontent.com/eswarasai/social-login/master/img/twitter-white.png') no-repeat scroll 5px 1px / 45px 45px padding-box border-box;
```
Comment (pavsorab) [inline frontend/asfui/core/static/assets/css/material-dashboard.css:18976]: It is not recommended to use images from github accounts which are not managed by us. The possible issues could be :

- License issues
- Any maintainers can upload malicious code to images
- maintenance issue (What if user deletes those images?)

Please add required images to our ASF repo.

**Review by pavsorab — CHANGES_REQUESTED** [review]

File: frontend/asfui/app/management/commands/Jira_Conf.py [hunk 22]
```diff
@@ -0,0 +1,92 @@
...
+def jira_status(ticket_num):
+    issue = jira.issue(ticket_num)
+    status = issue.fields.status
+    return status    
+
+def create_issue(query):
+    if settings.JIRA_ENABLED:
+        username = urllib.parse.quote_plus(settings.MONGO_USER)
+        password = urllib.parse.quote_plus(settings.MONGO_PASSWORD)
+        url = settings.MONGO_URL
```
[the remark this answers is not in the document]
Comment (psc4re) [reply frontend/asfui/app/management/commands/Jira_Conf.py:45]: will be updated in next MR.

File: frontend/asfui/app/management/commands/Jira_Conf.py [hunk 23]
```diff
@@ -0,0 +1,92 @@
...
+            query = {'_id': id}
+            mycol.update_one(query, {"$set":{"verified":True, "ignored":False,  "jira_created":True, "jira_closed":False, "jira_ticket":ticket_num, "jira_url": settings.JIRA_URL+"/browse/"+ticket_num}})
+    #if setting.EMAIL_ENABLED:
+    return
+        
+ 
+def ignore_issue(query):
+    username = urllib.parse.quote_plus(settings.MONGO_USER)
+    password = urllib.parse.quote_plus(settings.MONGO_PASSWORD)
+    url = settings.MONGO_URL
```
[the remark this answers is not in the document]
Comment (psc4re) [reply frontend/asfui/app/management/commands/Jira_Conf.py:64]: will be updated in next MR

File: frontend/asfui/app/management/commands/Jira_Conf.py (hunk 2 above)
Comment (psc4re) [reply frontend/asfui/app/management/commands/Jira_Conf.py:71]: debug added

File: frontend/asfui/authentication/views.py (hunk 13 above)
Comment (psc4re) [reply frontend/asfui/authentication/views.py:31]: Error will be displayed. so not required.

File: frontend/asfui/authentication/views.py (hunk 14 above)
Comment (psc4re) [reply frontend/asfui/authentication/views.py:33]: error displayed not required.

File: frontend/asfui/core/settings.py (hunk 15 above)
Comment (psc4re) [reply frontend/asfui/core/settings.py:251]: configurable setting added

**Comment by psc4re** [comment]
@pavsorab latest commit resolves the issues identified.

**Review by pavsorab — APPROVED** [review]
Changes looks fine to me.

**Review by pavsorab — APPROVED** [review]
Changes look fine to me.

**Pull request merged**